R/aggregate_game_stats_def.R

```diff
@@ -0,0 +1,580 @@
+
+  # combine all the stats together
+
+  player_df <- tackle_df %>%
+    dplyr::full_join(tackle_yards_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(pressure_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(int_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(safety_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_df_own, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_df_opp, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_yds_own_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_yds_opp_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(penalty_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(touchdown_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::mutate_if(is.numeric, tidyr::replace_na, 0) %>%
+    dplyr::left_join(
+      nflreadr::load_players() %>%
+        dplyr::select(
+          "player_id" = "gsis_id",
+          "player_display_name" = "display_name",
+          "player_name" = "short_name",
+          "position",
+          "position_group",
+          "headshot_url" = "headshot"
+        ),
+      by = "player_id"
+    ) %>%
+    dplyr::select(tidyselect::any_of(c(
+
+      # game information
+      "season",
+      "week",
+
+      # id information
+      "player_id",
+      "player_name",
+      "player_display_name",
+      "position",
+      "position_group",
+      "headshot_url",
+      "team",
+
+      # tackle stats
+      "def_tackles" = "tackles",
+      "def_tackles_solo" = "tackles_solo",
+      "def_tackles_with_assist" = "tackles_with_assist",
+      "def_tackle_assists" = "tackle_assists",
+      "def_tackles_for_loss" = "tackles_for_loss",
+      "def_tackles_for_loss_yards" = "tfl_yards",
+      "def_fumbles_forced" = "forced_fumbles",
+
+      # pressure stats
+      "def_sacks"="sacks",
+      "def_sack_yards"="sack_yards",
+      "def_qb_hits"="qb_hit",
+
+      # coverage stats
+      "def_interceptions"="int",
+      "def_interception_yards"="int_yards",
+      "def_pass_defended"="pass_defended",
+
+      # misc stats
+      "def_tds"="td",
+      "def_fumbles"="fumble",
+      "def_fumble_recovery_own"="fumble_recovery_own",
+      "def_fumble_recovery_yards_own"="fumble_recovery_yards_own",
+      "def_fumble_recovery_opp"="fumble_recovery_opp",
+      "def_fumble_recovery_yards_opp"="fumble_recovery_yards_opp",
+      "def_safety"="safety",
+      "def_penalty"="penalty",
+      "def_penalty_yards"="penalty_yards"
+    ))) %>%
+    dplyr::filter(!is.na(.data$player_id)) %>%
+    dplyr::arrange(.data$player_id, .data$season, .data$week)
+
+  # if user doesn't want week-by-week input, aggregate the whole df
+  if (isFALSE(weekly)) {
+    player_df <- player_df %>%
+      dplyr::group_by(.data$player_id, .data$team) %>%
```

Comment:
This group by team will cause different behavior compared to offense stats where we summarise by player only. Function will output multiple rows for one player if the player switched team during the season.

Reply:
I dislike this actual behaviour but in practice consistent with offense is fine. Also I never use this so I should not complain 😂